temporian/test/event_conversion_test.py

```diff
@@ -0,0 +1,183 @@
+        df = pd.DataFrame(
+            [
+                [666964, 1.0, 740.0],
+                [666964, 2.0, 508.0],
+                [574016, 3.0, 573.0],
+            ],
+            columns=["product_id", "timestamp", "costs"],
+        )
+
+        numpy_sampling = NumpySampling(
+            data={
+                (): np.array([1.0, 2.0, 3.0]),
+            },
+            index=[],
+        )
+
+        expected_numpy_event = NumpyEvent(
+            data={
+                (): [
+                    NumpyFeature(
+                        data=np.array([666964, 666964, 574016]),
+                        name="product_id",
+                    ),
+                    NumpyFeature(
+                        data=np.array([740.0, 508.0, 573.0]), name="costs"
+                    ),
+                ],
+            },
+            sampling=numpy_sampling,
+        )
+
+        numpy_event = NumpyEvent.from_dataframe(
+            df, index_names=[], timestamp_name="timestamp"
+        )
+
+        # validate
+        self.assertEqual(
+            True,
+            numpy_event == expected_numpy_event,
+        )
+
+    def test_numpy_event_to_df(self) -> None:
+        numpy_sampling = NumpySampling(
+            data={
+                (666964,): np.array([1.0, 2.0]),
+                (574016,): np.array([3.0]),
+            },
+            index=["product_id"],
+        )
+
+        numpy_event = NumpyEvent(
+            data={
+                (666964,): [
+                    NumpyFeature(data=np.array([740.0, 508.0]), name="costs")
+                ],
+                (574016,): [NumpyFeature(data=np.array([573.0]), name="costs")],
+            },
+            sampling=numpy_sampling,
+        )
+
+        expected_df = pd.DataFrame(
+            [
+                [666964, 1.0, 740.0],
+                [666964, 2.0, 508.0],
+                [574016, 3.0, 573.0],
+            ],
+            columns=["product_id", "timestamp", "costs"],
+        ).set_index(["product_id", "timestamp"])
+
+        df = numpy_event.to_dataframe()
+
+        # validate
+        self.assertEqual(
+            True,
+            df.equals(expected_df),
+        )
+
+    def test_numpy_event_to_df_no_index(self) -> None:
+        numpy_sampling = NumpySampling(
+            data={
+                (): np.array([1.0, 2.0, 3.0]),
+            },
+            index=[],
+        )
+
+        numpy_event = NumpyEvent(
+            data={
+                (): [
+                    NumpyFeature(
+                        data=np.array([666964, 666964, 574016]),
+                        name="product_id",
+                    ),
+                    NumpyFeature(
+                        data=np.array([740.0, 508.0, 573.0]), name="costs"
+                    ),
+                ],
+            },
+            sampling=numpy_sampling,
+        )
+
+        expected_df = pd.DataFrame(
+            [
+                [666964, 1.0, 740.0],
```

Comment:
Idea of things to test:
- when the timestamp is a Date.
- when one of the feature is a string.
- when one of the feature contains missing values .

Reply:
Should we support Date format in timestamp even if it's not in `DTYPE_MAPPING`?